code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -978,6 +988,32 @@
 	set category = "Robot Commands"
 	set_autosay()
 
+/mob/living/silicon/robot/proc/control_headlamp()
+	if(stat)
+		src << "<span class='danger'>This function is currently offline.</span>"
+		return
+
+//Some sort of magical "modulo" thing which somehow increments lamp power by 2, until it hits the max and resets to 0.
+	lamp_intensity = (lamp_intensity+2) % (lamp_max+2)

```

Comment:
Please do not do that to me again, RR. Math is evil and scary. I would rather re-write Changeling.